### PR TITLE
Update Golang version

### PR DESCRIPTION
**Pull request by mchompalova**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,35 +22,30 @@ jobs:
       - run:
           name: External Dependencies
           command: |
-            go get -u github.com/mattn/goveralls
-            go get -u github.com/jstemmer/go-junit-report
-            go get -u github.com/kardianos/govendor
-            go get -u github.com/haya14busa/goverage
+            GO111MODULE=off go get -u github.com/mattn/goveralls
+            GO111MODULE=off go get -u github.com/jstemmer/go-junit-report
+            curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.18.0
```
Comment (MiroslavGatsanoga) [inline .circleci/config.yml:27]: Since we are downloading the linter we need to add a step to run it below.

File: Dockerfile [hunk 2]
```diff
@@ -1,33 +1,26 @@
...
+  && LDFLAGS="-s -w -X '"${BUILDINFO_PACKAGE}$VERSION"' -X '"${BUILDINFO_PACKAGE}$DATETIME"' -X '"${BUILDINFO_PACKAGE}$REPOSITORY"' -X '"${BUILDINFO_PACKAGE}$REVISION"' -X '"${BUILDINFO_PACKAGE}$BUILDER"'" \
   && echo "Build flags: $LDFLAGS" \
-  && echo "Fetching dependencies..." \
-  && go get -u github.com/kardianos/govendor \
-  && $GOPATH/bin/govendor sync \
-  && go build -ldflags="${LDFLAGS}" \
-  && mv ${PROJECT} /${PROJECT} \
-  && apk del .build-dependencies \
-  && rm -rf $GOPATH /var/cache/apk/*
+  && CGO_ENABLED=0 go build -a -o /artifacts/${PROJECT} -ldflags="${LDFLAGS}"
```
Comment (MiroslavGatsanoga) [inline Dockerfile:15]: Pass `-mod=readonly` in order to prevent accidental modification of the go.mod or go.sum files.

File: README.md [hunk 3]
```diff
@@ -6,27 +6,31 @@ This tool can back up or restore MongoDB collections while DB is running to/from
 
 You can deploy a docker container that will run backups on schedule. Or you can just run the container to make a single backup, or restore from a given point of time.
 
-For the schedule option, the state of backups is kept in a boltdb file. (at `/var/data/mongo-hot-backup/state.db` or where you set it)
+For the schedule option, the state of backups is kept in a boltdb file at `/var/data/mongo-hot-backup/state.db` or where you set it.
 
-Health endpoint is available at `0.0.0.0:8080/__health` and will report healthy if there was a successful backup for each configured collection in the last X hours, also configurable. Good-to-go `/__gtg` endpoint available as well, and `/build-info`.
+A health endpoint is available at `0.0.0.0:8080/__health` and will report healthy if there was a successful backup for each configured collection in the last X hours, also configurable. Good-to-go `/__gtg` endpoint available as well, and `/build-info`.
```
Comment (MiroslavGatsanoga) [inline README.md:11]: The build info endpoint is `__/build-info`.

**Review by MiroslavGatsanoga — COMMENTED** [review]
Note that the latest release is `2.0.2` so this one should be tagged `v2.0.3` since it is a fix. 
Technically since the transition to modules is made here and this is a v2+ project it is recommended to bump the major version number i.e. `v3.0.0` and append `/v3` to the module path, but since the whole codebase is contained in the `main` package it is fine to stick with `v2.0.3`.

File: .circleci/config.yml (hunk 1 above)
Comment (ekaneva) [reply .circleci/config.yml:27]: Updated the `.circleci/config.yml` to using the `ft/golang-ci` orb.

**Comment by ekaneva** [comment]
Due to the migration to Go modules and the recommended bumping of the version in this case to `v3.0.0`, I renamed the branch to `major/UPPSF-666...`

This automatically closed this PR but here is a link to the new one: https://github.com/Financial-Times/mongo-hot-backup/pull/19, where all of the above comments were addressed.

File: README.md (hunk 3 above)
Comment (ekaneva) [reply README.md:11]: Renamed to `/__build-info`.

**Pull request closed without merging**